File: packages/maplibre/android/src/main/java/io/capawesome/capacitorjs/plugins/maplibre/classes/Padding.java
package io.capawesome.capacitorjs.plugins.maplibre.classes;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;

/**
 * The padding between the content of a map and its edges in CSS pixels.
 */
public class Padding {

    private final double bottom;
    private final double left;
    private final double right;
    private final double top;

    public Padding(@Nullable JSObject object) {
        this.bottom = object == null ? 0 : object.optDouble("bottom", 0);
        this.left = object == null ? 0 : object.optDouble("left", 0);
        this.right = object == null ? 0 : object.optDouble("right", 0);
        this.top = object == null ? 0 : object.optDouble("top", 0);
    }

    public double getBottom() {
        return bottom;
    }

    public double getLeft() {
        return left;
    }

    public double getRight() {
        return right;
    }

    public double getTop() {
        return top;
    }
}

File: packages/maplibre/ios/Plugin/Classes/Padding.swift
import Capacitor
import Foundation
import UIKit

public class Padding {
    let bottom: Double
    let left: Double
    let right: Double
    let top: Double

    init(_ object: JSObject?) {
        self.bottom = MapLibreHelper.getDouble(object, "bottom") ?? 0
        self.left = MapLibreHelper.getDouble(object, "left") ?? 0
        self.right = MapLibreHelper.getDouble(object, "right") ?? 0
        self.top = MapLibreHelper.getDouble(object, "top") ?? 0
    }

    func toEdgeInsets() -> UIEdgeInsets {
        return UIEdgeInsets(top: top, left: left, bottom: bottom, right: right)
    }
}
